src/common/websocket/api.ts

```diff
@@ -71,6 +58,44 @@ export default class DerivWS {
         }
     }
 
+    private messageHandler = (msg: { data: string }) => {
+        const response: DerivAPIResponse = JSON.parse(msg.data)
+        const reqId = response.req_id
+
+        if (reqId && reqId in this.pendingRequests) {
+            this.pendingRequests[reqId].resolve(response)
+            delete this.pendingRequests[reqId]
+        }
+    }
+
+    private recursiveConnect = () => {
+        this.connection = new WebSocket(this.socket_url)
+        this.connection.onopen = this.openHandler
+        this.connection.onmessage = this.messageHandler
+        this.connection.onclose = () => {
+            let lastActionTaken = new Date().getTime()
+            const checkLastAction = () => {
+                const now = new Date().getTime()
+                if (now - lastActionTaken > 500) {
```

Comment:
@NikitK-deriv please elaborate on why do we need to check for the last action taken and why do we need to use 500 here? if we want to reopen connection when it's closed, we can simply do this:
```suggestion
                this.connection.onclose = () => {
                    this.recursiveConnect()
                }
```
I believe we don't need any listeners here especially while we'll need to remove them too at some point, otherwise they will fire multiple number of times whenever each of the below events happens. If the connection is closed, it's obviously closed because the user was idle and didn't do any actions, right? Or are there any cases when the connection gets closed and we don't want to reopen it?
Also, I think the PR description is a bit out-dated, you are still saying that you're going to reload the page. Please check 🙂

Reply:
Thanks for note 🙇